By Nishant Limbachia <nishant@mnspace.net>

Patch awstats_configure.pl file in case somebody runs it.
Following things are patched:
1. Use correct paths 
2. Stop awstats_configure.pl to get the current working directory
   on first run (use standard path instead)
3. Added support to restart httpd server for Slackware

--- tools/awstats_configure.pl.orig	2009-01-27 13:23:05.000000000 -0600
+++ tools/awstats_configure.pl	2009-01-27 13:36:46.000000000 -0600
@@ -28,16 +28,14 @@
 $AWSTATS_MODEL_CONFIG
 $AWSTATS_DIRDATA_PATH
 /;
-$AWSTATS_PATH='';
-$AWSTATS_ICON_PATH='/usr/local/awstats/wwwroot/icon';
-$AWSTATS_CSS_PATH='/usr/local/awstats/wwwroot/css';
-$AWSTATS_CLASSES_PATH='/usr/local/awstats/wwwroot/classes';
-$AWSTATS_CGI_PATH='/usr/local/awstats/wwwroot/cgi-bin';
+$AWSTATS_PATH='@DOCROOT@/awstats';
+$AWSTATS_ICON_PATH='$AWSTATS_PATH/wwwroot/icon';
+$AWSTATS_CSS_PATH='$AWSTATS_PATH/wwwroot/css';
+$AWSTATS_CLASSES_PATH='$AWSTATS_PATH/wwwroot/classes';
+$AWSTATS_CGI_PATH='$AWSTATS_PATH/wwwroot/cgi-bin';
 $AWSTATS_MODEL_CONFIG='/etc/awstats/awstats.model.conf';		# Used only when configure ran on linux
 $AWSTATS_DIRDATA_PATH='/var/lib/awstats';						# Used only when configure ran on linux
 
-
-
 #-------------------------------------------------------
 # Defines
 #-------------------------------------------------------
@@ -306,7 +304,7 @@
 print "\n-----> Running OS detected: $OSLib{$OS}\n";
 
 if ($OS eq 'linux') {
-	$AWSTATS_PATH=`pwd`; $AWSTATS_PATH =~ s/[\r\n]//;
+	$AWSTATS_PATH =~ s/[\r\n]//;
 	$AWSTATS_PATH=~s/tools[\\\/]?$//;
 	$AWSTATS_PATH=~s/[\\\/]$//;
 	if ($AWSTATS_PATH ne '/usr/local/awstats') {
@@ -662,6 +660,12 @@
     		print "\n-----> Restart Web server with '$command'\n";
 	 	    my $ret=`$command`;
 	 	    print "$ret";
+	} if (-f "/etc/slackware-version") {
+	    # We are on Slackware :)
+		my $command="/etc/rc.d/rc.httpd restart";
+		print "\n-----> Restart Web server with '$command'\n";
+		    my $ret=`$command`;
+		    print "$ret";
         } elsif (-x "/sbin/service") {
             # We are not on debian
        	 	my $command="/sbin/service httpd restart";
